src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -0,0 +1,527 @@
+ *
+ * deparse_function_stmts.c
+ *
+ * Copyright (c) 2019, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/htup_details.h"
+#include "catalog/namespace.h"
+#include "catalog/pg_proc.h"
+#include "commands/defrem.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/commands.h"
+#include "distributed/deparser.h"
+#include "distributed/version_compat.h"
+#include "lib/stringinfo.h"
+#include "nodes/makefuncs.h"
+#include "nodes/nodes.h"
+#include "nodes/value.h"
+#include "parser/parse_func.h"
+#include "parser/parse_type.h"
+#include "utils/builtins.h"
+#include "utils/guc.h"
+#include "utils/lsyscache.h"
+#include "utils/syscache.h"
+
+/* forward declaration for deparse functions */
+static void AppendAlterFunctionStmt(StringInfo buf, AlterFunctionStmt *stmt);
+static void AppendDropFunctionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendFunctionName(StringInfo buf, ObjectWithArgs *func);
+static void AppendFunctionNameList(StringInfo buf, List *objects);
+
+static void AppendDefElem(StringInfo buf, DefElem *def);
+static void AppendDefElemStrict(StringInfo buf, DefElem *def);
+static void AppendDefElemVolatility(StringInfo buf, DefElem *def);
+static void AppendDefElemLeakproof(StringInfo buf, DefElem *def);
+static void AppendDefElemSecurity(StringInfo buf, DefElem *def);
+static void AppendDefElemParallel(StringInfo buf, DefElem *def);
+static void AppendDefElemCost(StringInfo buf, DefElem *def);
+static void AppendDefElemRows(StringInfo buf, DefElem *def);
+static void AppendDefElemSet(StringInfo buf, DefElem *def);
+
+static void AppendRenameFunctionStmt(StringInfo buf, RenameStmt *stmt);
+static void AppendAlterFunctionSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt);
+static void AppendAlterFunctionOwnerStmt(StringInfo buf, AlterOwnerStmt *stmt);
+static void AppendAlterFunctionDependsStmt(StringInfo buf, AlterObjectDependsStmt *stmt);
+
+
+const char *
+DeparseAlterFunctionStmt(AlterFunctionStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	AppendAlterFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseRenameFunctionStmt(RenameStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->renameType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->renameType == OBJECT_FUNCTION || stmt->renameType == OBJECT_PROCEDURE);
+#endif
+
+	AppendRenameFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseAlterFunctionSchemaStmt(AlterObjectSchemaStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->objectType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionSchemaStmt(&str, stmt);
```

Comment:
Minor: In general, I think it'd be easier to move `AppendXXXStmt ` right after `DeparseXXXStmt `. Otherwise, the readers need to navigate to see the actual logic.

Reply:
I ordered them locally. I hope it will work out this time.